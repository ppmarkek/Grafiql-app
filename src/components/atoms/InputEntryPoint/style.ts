import { Button, Grid, Tab, TextField } from '@mui/material';
import { styled } from '@mui/system';

export const StyledTextField = styled(TextField)({
  height: '100%',
  width: '100%',
  '.MuiInputBase-root': {
    height: '100%',
  },
  textarea: {
    color: '#fff',
    height: '100% !important',
  },
});

export const ButtonGrid = styled(Grid)({
  width: '7%',
  background: 'rgba(0, 0, 0, 0.06)',
  justifyContent: 'flex-start',
  alignItems: 'center',
  flexDirection: 'column',
  paddingTop: '15px',
  gap: '15px',
});

export const ButtonStart = styled(Button)({
  all: 'unset',
  width: '40px',
  height: '40px',
  background: '#FF5794',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: '8px',
  cursor: 'pointer',
  border: '1px solid #FF5794',
  transition: 'all 0.5s',

  span: {
    display: 'none',
  },
});

export const StyledButton = styled(Button)({
  all: 'unset',
  width: '42px',
  height: '42px',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: '8px',
  cursor: 'pointer',
  transition: 'all 0.5s',

  span: {
    display: 'none',
  },
});

export const GridTabs = styled(Grid)({
  flexDirection: 'column',
  backgroundColor: 'rgba(0, 0, 0, 0.09)',
  borderBottomLeftRadius: '4px',
  borderBottomRightRadius: '4px',
  height: '25%',
});

export const StyledTab = styled(Tab)({
  color: '#fff',
});
