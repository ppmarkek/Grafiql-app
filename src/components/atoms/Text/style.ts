import { Typography } from '@mui/material';
import { styled } from '@mui/system';

type TextStyleProps = {
  $small?: boolean;
  $color?: string;
};

export const H1 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '34px',
  lineHeight: '42px',
  color: $color || '#000',
}));

export const H2 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '28px',
  lineHeight: '38px',
  color: $color || '#000',
}));

export const H3 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '25px',
  lineHeight: '32px',
  color: $color || '#000',
}));

export const H4 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '23px',
  lineHeight: '27px',
  color: $color || '#000',
}));

export const H5 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '21px',
  lineHeight: '24px',
  color: $color || '#000',
}));

export const Bold = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color' && prop !== '$small',
})(({ $color, $small }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: $small ? '16px' : '18px',
  color: $color || '#000',
  lineHeight: '21px',
}));

export const Regular = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color' && prop !== '$small',
})(({ $color, $small }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '400',
  fontSize: $small ? '16px' : '18px',
  lineHeight: '21px',
  color: $color || '#000',
}));

export const Error = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$small',
})(({ $small }: TextStyleProps) => ({
  color: 'rgb(211, 47, 47)',
  fontFamily: 'Heebo, sansSerif',
  fontWeight: '400',
  fontSize: $small ? '16px' : '18px',
  lineHeight: '21px',
}));
