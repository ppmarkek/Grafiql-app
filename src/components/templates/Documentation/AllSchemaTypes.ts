export const AllSchemaTypes = [
  'String',
  'Int',
  'FilmsConnection',
  'PageInfo',
  'Boolean',
  'FilmsEdge',
  'Film',
  'Node',
  'ID',
  'FilmSpeciesConnection',
  'FilmSpeciesEdge',
  'Species',
  'Float',
  'Planet',
  'PlanetResidentsConnection',
  'PlanetResidentsEdge',
  'Person',
  'PersonFilmsConnection',
  'PersonFilmsEdge',
  'PersonStarshipsConnection',
  'PersonStarshipsEdge',
  'Starship',
  'StarshipPilotsConnection',
  'StarshipPilotsEdge',
  'StarshipFilmsConnection',
  'StarshipFilmsEdge',
  'PersonVehiclesConnection',
  'PersonVehiclesEdge',
  'Vehicle',
  'VehiclePilotsConnection',
  'VehiclePilotsEdge',
  'VehicleFilmsConnection',
  'VehicleFilmsEdge',
  'PlanetFilmsConnection',
  'PlanetFilmsEdge',
  'SpeciesPeopleConnection',
  'SpeciesPeopleEdge',
  'SpeciesFilmsConnection',
  'SpeciesFilmsEdge',
  'FilmStarshipsConnection',
  'FilmStarshipsEdge',
  'FilmVehiclesConnection',
  'FilmVehiclesEdge',
  'FilmCharactersConnection',
  'FilmCharactersEdge',
  'FilmPlanetsConnection',
  'FilmPlanetsEdge',
  'PeopleConnection',
  'PeopleEdge',
  'PlanetsConnection',
  'PlanetsEdge',
  'SpeciesConnection',
  'SpeciesEdge',
  'StarshipsConnection',
  'StarshipsEdge',
  'VehiclesConnection',
  'VehiclesEdge',
];
