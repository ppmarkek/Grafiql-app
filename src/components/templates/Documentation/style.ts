import { Button, Grid } from '@mui/material';
import { styled } from '@mui/system';

type ButtonProps = {
  active?: boolean;
};

export const Wrapper = styled(Grid)({
  height: '100%',
  width: '30px',
});

export const OpenGrid = styled(Grid)({
  height: '100%',
  width: '100%',
  borderRight: '1px solid #444',
});

export const OpenButton = styled(Button)(({ active }: ButtonProps) => ({
  all: 'unset',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  background: '#000',
  borderRadius: '20px',
  width: '25px',
  height: '25px',
  cursor: 'pointer',
  position: 'absolute',
  left: active ? '387px' : '17px',
  transform: active ? 'rotate(0.5turn)' : 'none',
  top: '220px',
  zIndex: '1201',
  transition: 'all 0.2s',
  span: {
    display: 'none',
  },
  ':hover': {
    background: '#6B59CC',
  },
}));

export const DrawerGrid = styled(Grid)({
  display: 'flex',
  gap: '30px',
  flexDirection: 'column',
  width: '400px',
  padding: '20px',
});
