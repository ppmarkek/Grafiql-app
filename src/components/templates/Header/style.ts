import { Grid } from '@mui/material';
import { Link } from 'react-router-dom';
import { styled } from '@mui/system';

type LinkStyleProps = {
  $width?: string;
  $isSticky?: boolean;
};

export const Wrapper = styled(Grid)(({ $isSticky }: LinkStyleProps) => ({
  height: $isSticky ? '80px' : '90px',
  backgroundColor: $isSticky ? '#f0f0f0' : 'transparent',
  justifyContent: 'space-between',
  alignItems: 'center',
  padding: '0px 50px',
  position: 'sticky',
  top: '0',
  zIndex: '10',
  transition: 'all 0.5s ease-in-out',
}));

export const StyledLink = styled(Link)(({ $width }: LinkStyleProps) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  width: $width || 'auto',
  textDecoration: 'none',
  color: '#000',
  fontWeight: '700',
  fontSize: '18px',
  padding: '10px',
  borderRadius: '10px',
  transition: 'all 0.5s',

  '&:hover': {
    boxShadow:
      '0 -10px 25px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.1)',
  },
}));
