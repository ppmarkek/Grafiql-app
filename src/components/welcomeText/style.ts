import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  background-color: #fff;
`;

export const Heading = styled.h1`
  font-size: 24px;
  font-weight: bold;
  color: #fff;
  margin-bottom: 1.2rem;
`;

export const Paragraph = styled.p`
  font-size: 16px;
  line-height: 1.5;
  color: #ddd;
  margin-bottom: 0.6rem;
`;
