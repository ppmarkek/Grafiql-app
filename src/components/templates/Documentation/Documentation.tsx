import { useState } from 'react';
import { OpenButton, OpenGrid, Wrapper } from './style';
import { Drawer } from '@mui/material';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';

const Documentation = () => {
  const [state, setState] = useState(false);

  return (
    <Wrapper>
      <OpenGrid>
        <OpenButton>
          <ChevronRightIcon />
        </OpenButton>
      </OpenGrid>
      <Drawer anchor={'left'} open={state}></Drawer>
    </Wrapper>
  );
};

export default Documentation;
