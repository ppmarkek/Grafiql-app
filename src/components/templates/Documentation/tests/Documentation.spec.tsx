import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Documentation from '../Documentation';
import { OpenButton } from '../style';

describe('Documentation Component', () => {
  it('renders without crashing', () => {
    render(<Documentation />);
  });

  it('toggles drawer state on button click', () => {
    render(<Documentation />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Documentation />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('initially has the drawer closed and type set to all', async () => {
    render(<Documentation />);
    expect(screen.queryByTestId('drawer')).toBeNull();
  });

  it('displays correct content when a schema type is selected', async () => {
    render(<Documentation />);
    fireEvent.click(screen.getByTestId('open-button'));
    const queryRootButton = await screen.findByText('query: Root');
    fireEvent.click(queryRootButton);
    expect(screen.getByText('Root')).toBeInTheDocument();
  });

  it('returns to all types view when NavigateBeforeIcon is clicked', async () => {
    render(<Documentation />);
    fireEvent.click(screen.getByTestId('open-button'));
    const queryRootButton = await screen.findByText('query: Root');
    fireEvent.click(queryRootButton);
    fireEvent.click(screen.getByTestId('NavigateBeforeIcon'));
    expect(screen.getByText('All Schema Types')).toBeInTheDocument();
  });

  it('applies correct styles when active is true', () => {
    render(<OpenButton active="true" />);
    const button = screen.getByRole('button');
    expect(button).toHaveStyle('left: 387px');
    expect(button).toHaveStyle('transform: rotate(0.5turn)');
  });

  it('applies correct styles when active is false', () => {
    render(<OpenButton active={false} />);
    const button = screen.getByRole('button');
    const style = window.getComputedStyle(button);
    expect(style.left).toBe('17px');
    expect(style.transform).toBe('none');
  });
});
