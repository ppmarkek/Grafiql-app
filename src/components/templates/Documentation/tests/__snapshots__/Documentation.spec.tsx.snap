// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root css-ldngnm-MuiGrid-root"
  >
    <div
      class="MuiGrid-root css-xmbyct-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1332u5p-MuiButtonBase-root-MuiButton-root"
        data-testid="open-button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
